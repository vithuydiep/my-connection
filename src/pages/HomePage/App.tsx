import { useEffect, useState } from "react";
import axios from "axios";
import { formatUsername } from "./../../Utils/utils";

interface UserItem {
  id: string;
  name: string;
  username: string;
}

export default function App() {
  const [users, setUsers] = useState([]);

  // Load the data from the server
  useEffect(() => {
    const getUsers = async () => {
      const response = await axios.get(
        "https://jsonplaceholder.typicode.com/users"
      );
      setUsers(response.data);
    };
    getUsers();
  }, []);

  return (
    <div>
      <h1>User List</h1>
      {users.map((item: UserItem) => (
        <li key={item.id}>{item.username}</li>
      ))}
    </div>
  );
}
