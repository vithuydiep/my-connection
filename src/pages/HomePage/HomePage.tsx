import { Pagination } from '@mui/material';
import { orderBy } from 'lodash';
import { useState } from 'react';
import { Sidebar } from '../../Components';
import Footer from '../../Components/Footer/Footer';
import HeaderItem from '../../Components/HeaderItem/HeaderItem';
import Header from '../../Layout/Header/Header';
import { AccordianIcon } from '../../assets/svgs';
import useHomePageStyle from './HomePageStyles';
import { useNavigate } from 'react-router-dom';
import { ROUTE_PATHS } from '../../Routings/constant';




interface DataConnectionProps {
  id: number;
  connectionName: string;
  connectionType: string;
  status: string;
  createdBy: string;
  creationTime: string;
  lastUpdate: string;
}

const initData: DataConnectionProps[] = [
  {
    id: 1,
    connectionName: ' W4',
    connectionType: 'Oracle',
    status: 'Succeeded',
    createdBy: 'vi.tran',
    creationTime: '2/5/2022-10:00 am',
    lastUpdate: '2/5/2022-10:00 am',
  },
  {
    id: 2,
    connectionName: 'T24',
    connectionType: 'Oracle',
    status: 'Failed',
    createdBy: 'vi.tran',
    creationTime: '2/5/2022-10:00 am',
    lastUpdate: '2/5/2022-10:00 am',
  },
  {
    id: 3,
    connectionName: 'T24',
    connectionType: 'Oracle',
    status: 'N/A',
    createdBy: 'vi.tran',
    creationTime: '2/5/2022-10:00 am',
    lastUpdate: '2/5/2022-10:00 am',
  },
];

function HomePage() {
  const navigate = useNavigate()
  const [data, setData] = useState<DataConnectionProps[]>(initData);
  const [sorting, setSorting] = useState({
    connectionName: false,
    connectionType: false,
    status: false,
    createdBy: false,
    creationTime: false,
    lastUpdate: false,
  });

  const {
    wrapper,
    content,
    title,
    groupWrapper,
    disableBtn,
    contentWrapper,
    table,
    wrapTable,
    pagination,
  } = useHomePageStyle();

  const onCheckClass = (value: string): string => {
    switch (value) {
      case 'Succeeded':
        return 'success';
      case 'Failed':
        return 'fail';
      default:
        return '';
    }
  };

  const onHandleSorting = (name: string, value: boolean): void => {
    setSorting({
      ...sorting,
      [name]: value,
    });
    const newData = orderBy(data, [name], [value ? 'desc' : 'asc'])
    setData(newData)
  };


  return (
    <>
      <Header customize />
      <div className={wrapper}>
        <Sidebar />
        <div className={content}>
          <div className={title}>Connections</div>
          <div className={groupWrapper}>
            <p>Connections</p>
            <div>
              <button className={disableBtn}>
                Actions
                <AccordianIcon />
              </button>
              <button onClick={()=> {
                navigate(ROUTE_PATHS.createConnectionPage)
              }}>Create Connection</button>
            </div>
          </div>
          <div className={contentWrapper}>
            <div className={wrapTable}>
              <table className={table}>
                <thead>
                  <tr>
                    <th></th>
                    <th>
                      <HeaderItem
                        name='Connection Name'
                        sorting={sorting.connectionName}
                        onHandleSorting={onHandleSorting}
                        objectKey='connectionName'
                      />
                    </th>
                    <th>
                      <HeaderItem
                        name='Connection Type'
                        sorting={sorting.connectionType}
                        onHandleSorting={onHandleSorting}
                        objectKey='connectionType'
                      />
                    </th>
                    <th>
                      <HeaderItem
                        name='Status'
                        sorting={sorting.status}
                        onHandleSorting={onHandleSorting}
                        objectKey='status'
                      />
                    </th>
                    <th>
                      <HeaderItem
                        name='Created By'
                        sorting={sorting.createdBy}
                        onHandleSorting={onHandleSorting}
                        objectKey='createdBy'
                      />
                    </th>
                    <th>
                      <HeaderItem
                        name='Creation Time'
                        sorting={sorting.creationTime}
                        onHandleSorting={onHandleSorting}
                        objectKey='creationTime'
                      />
                    </th>
                    <th>
                      <HeaderItem
                        name='Last Updated'
                        sorting={sorting.lastUpdate}
                        onHandleSorting={onHandleSorting}
                        objectKey='lastUpdate'
                      />
                    </th>
                  </tr>
                </thead>
                <tbody>
                  {data.map((item) => (
                    <tr key={item.id}>
                      <td>
                        <input type='radio' name='check' />
                      </td>
                      <td className='textPrimary'>{item.connectionName}</td>
                      <td>{item.connectionType}</td>
                      <td className={onCheckClass(item.status)}>
                        {item.status}
                      </td>
                      <td>{item.createdBy}</td>
                      <td>{item.creationTime}</td>
                      <td>{item.lastUpdate}</td>
                    </tr>
                  ))}
                </tbody>
              </table>
            </div>
          </div>
          <Pagination
            count={10}
            variant='outlined'
            shape='rounded'
            className={pagination}
            color='primary'
          />
        </div>
      </div>
      <Footer />
    </>
  );
}

export default HomePage;
