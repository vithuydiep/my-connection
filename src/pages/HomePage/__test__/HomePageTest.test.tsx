/* eslint-disable testing-library/no-wait-for-snapshot */
/* eslint-disable testing-library/no-render-in-setup */
import { cleanup, render, waitFor , screen, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import HomePage from '../HomePage';


const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useNavigate: () => mockedUsedNavigate,
}));

const Component = () => (
  <BrowserRouter>
    <HomePage />
  </BrowserRouter>
);

describe('home page', () => {
  beforeEach(async () => {
    render(<Component />);
  });

  afterEach(() => {
    cleanup();
  });

  it('should render and match the snapshot', async () => {
    const { asFragment } = render(<Component />);
    await waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    });
  });

  it('navigate will be call when click create new connection', async () => {
    const createBtn = screen.getByText('Create Connection')
    fireEvent.click(createBtn)
    expect(mockedUsedNavigate).toBeCalled()
  });
});
