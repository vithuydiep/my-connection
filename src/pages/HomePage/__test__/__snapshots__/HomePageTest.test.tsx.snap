// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home page should render and match the snapshot 1`] = `
<DocumentFragment>
  <header
    class="header-0-2-10"
  >
    <a
      href="/"
    >
      <img
        alt=""
        class="logo-0-2-15"
        src="logo.png"
      />
    </a>
    <div
      class="rightContent-0-2-12"
    >
      <p
        class="username-0-2-13"
      >
        User Name
      </p>
      <img
        alt=""
        class="avatar-0-2-11"
        src="my-avatar.jpg"
      />
      <button
        class="settingButton-0-2-14"
        type="button"
      >
        <svg
          height="20px"
          viewBox="0 0 50 50"
          width="20px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 22.205078 2 A 1.0001 1.0001 0 0 0 21.21875 2.8378906 L 20.246094 8.7929688 C 19.076509 9.1331971 17.961243 9.5922728 16.910156 10.164062 L 11.996094 6.6542969 A 1.0001 1.0001 0 0 0 10.708984 6.7597656 L 6.8183594 10.646484 A 1.0001 1.0001 0 0 0 6.7070312 11.927734 L 10.164062 16.873047 C 9.583454 17.930271 9.1142098 19.051824 8.765625 20.232422 L 2.8359375 21.21875 A 1.0001 1.0001 0 0 0 2.0019531 22.205078 L 2.0019531 27.705078 A 1.0001 1.0001 0 0 0 2.8261719 28.691406 L 8.7597656 29.742188 C 9.1064607 30.920739 9.5727226 32.043065 10.154297 33.101562 L 6.6542969 37.998047 A 1.0001 1.0001 0 0 0 6.7597656 39.285156 L 10.648438 43.175781 A 1.0001 1.0001 0 0 0 11.927734 43.289062 L 16.882812 39.820312 C 17.936999 40.39548 19.054994 40.857928 20.228516 41.201172 L 21.21875 47.164062 A 1.0001 1.0001 0 0 0 22.205078 48 L 27.705078 48 A 1.0001 1.0001 0 0 0 28.691406 47.173828 L 29.751953 41.1875 C 30.920633 40.838997 32.033372 40.369697 33.082031 39.791016 L 38.070312 43.291016 A 1.0001 1.0001 0 0 0 39.351562 43.179688 L 43.240234 39.287109 A 1.0001 1.0001 0 0 0 43.34375 37.996094 L 39.787109 33.058594 C 40.355783 32.014958 40.813915 30.908875 41.154297 29.748047 L 47.171875 28.693359 A 1.0001 1.0001 0 0 0 47.998047 27.707031 L 47.998047 22.207031 A 1.0001 1.0001 0 0 0 47.160156 21.220703 L 41.152344 20.238281 C 40.80968 19.078827 40.350281 17.974723 39.78125 16.931641 L 43.289062 11.933594 A 1.0001 1.0001 0 0 0 43.177734 10.652344 L 39.287109 6.7636719 A 1.0001 1.0001 0 0 0 37.996094 6.6601562 L 33.072266 10.201172 C 32.023186 9.6248101 30.909713 9.1579916 29.738281 8.8125 L 28.691406 2.828125 A 1.0001 1.0001 0 0 0 27.705078 2 L 22.205078 2 z M 23.056641 4 L 26.865234 4 L 27.861328 9.6855469 A 1.0001 1.0001 0 0 0 28.603516 10.484375 C 30.066026 10.848832 31.439607 11.426549 32.693359 12.185547 A 1.0001 1.0001 0 0 0 33.794922 12.142578 L 38.474609 8.7792969 L 41.167969 11.472656 L 37.835938 16.220703 A 1.0001 1.0001 0 0 0 37.796875 17.310547 C 38.548366 18.561471 39.118333 19.926379 39.482422 21.380859 A 1.0001 1.0001 0 0 0 40.291016 22.125 L 45.998047 23.058594 L 45.998047 26.867188 L 40.279297 27.871094 A 1.0001 1.0001 0 0 0 39.482422 28.617188 C 39.122545 30.069817 38.552234 31.434687 37.800781 32.685547 A 1.0001 1.0001 0 0 0 37.845703 33.785156 L 41.224609 38.474609 L 38.53125 41.169922 L 33.791016 37.84375 A 1.0001 1.0001 0 0 0 32.697266 37.808594 C 31.44975 38.567585 30.074755 39.148028 28.617188 39.517578 A 1.0001 1.0001 0 0 0 27.876953 40.3125 L 26.867188 46 L 23.052734 46 L 22.111328 40.337891 A 1.0001 1.0001 0 0 0 21.365234 39.53125 C 19.90185 39.170557 18.522094 38.59371 17.259766 37.835938 A 1.0001 1.0001 0 0 0 16.171875 37.875 L 11.46875 41.169922 L 8.7734375 38.470703 L 12.097656 33.824219 A 1.0001 1.0001 0 0 0 12.138672 32.724609 C 11.372652 31.458855 10.793319 30.079213 10.427734 28.609375 A 1.0001 1.0001 0 0 0 9.6328125 27.867188 L 4.0019531 26.867188 L 4.0019531 23.052734 L 9.6289062 22.117188 A 1.0001 1.0001 0 0 0 10.435547 21.373047 C 10.804273 19.898143 11.383325 18.518729 12.146484 17.255859 A 1.0001 1.0001 0 0 0 12.111328 16.164062 L 8.8261719 11.46875 L 11.523438 8.7734375 L 16.185547 12.105469 A 1.0001 1.0001 0 0 0 17.28125 12.148438 C 18.536908 11.394293 19.919867 10.822081 21.384766 10.462891 A 1.0001 1.0001 0 0 0 22.132812 9.6523438 L 23.056641 4 z M 25 17 C 20.593567 17 17 20.593567 17 25 C 17 29.406433 20.593567 33 25 33 C 29.406433 33 33 29.406433 33 25 C 33 20.593567 29.406433 17 25 17 z M 25 19 C 28.325553 19 31 21.674447 31 25 C 31 28.325553 28.325553 31 25 31 C 21.674447 31 19 28.325553 19 25 C 19 21.674447 21.674447 19 25 19 z"
          />
        </svg>
      </button>
    </div>
  </header>
  <div
    class="wrapper-0-2-1"
  >
    <div
      class="wrapper-0-2-16"
    >
      <a
        aria-current="page"
        class="menuHomeActive-0-2-19 active"
        href="/"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-house icon-0-2-20"
          data-icon="house"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M575.8 255.5c0 18-15 32.1-32 32.1h-32l.7 160.2c0 2.7-.2 5.4-.5 8.1V472c0 22.1-17.9 40-40 40H456c-1.1 0-2.2 0-3.3-.1c-1.4 .1-2.8 .1-4.2 .1H416 392c-22.1 0-40-17.9-40-40V448 384c0-17.7-14.3-32-32-32H256c-17.7 0-32 14.3-32 32v64 24c0 22.1-17.9 40-40 40H160 128.1c-1.5 0-3-.1-4.5-.2c-1.2 .1-2.4 .2-3.6 .2H104c-22.1 0-40-17.9-40-40V360c0-.9 0-1.9 .1-2.8V287.6H32c-18 0-32-14-32-32.1c0-9 3-17 10-24L266.4 8c7-7 15-8 22-8s15 2 21 7L564.8 231.5c8 7 12 15 11 24z"
            fill="currentColor"
          />
        </svg>
        <p
          class="namePath-0-2-21"
        >
          Home
        </p>
      </a>
      <a
        aria-current="page"
        class="menuLinkActive-0-2-18"
        href="/"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-database icon-0-2-20"
          data-icon="database"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M448 80v48c0 44.2-100.3 80-224 80S0 172.2 0 128V80C0 35.8 100.3 0 224 0S448 35.8 448 80zM393.2 214.7c20.8-7.4 39.9-16.9 54.8-28.6V288c0 44.2-100.3 80-224 80S0 332.2 0 288V186.1c14.9 11.8 34 21.2 54.8 28.6C99.7 230.7 159.5 240 224 240s124.3-9.3 169.2-25.3zM0 346.1c14.9 11.8 34 21.2 54.8 28.6C99.7 390.7 159.5 400 224 400s124.3-9.3 169.2-25.3c20.8-7.4 39.9-16.9 54.8-28.6V432c0 44.2-100.3 80-224 80S0 476.2 0 432V346.1z"
            fill="currentColor"
          />
        </svg>
        <p
          class="namePath-0-2-21"
        >
          Connections
        </p>
      </a>
      <a
        class="menuLink-0-2-17"
        href="/feature2"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-user icon-0-2-20"
          data-icon="user"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224 256A128 128 0 1 0 224 0a128 128 0 1 0 0 256zm-45.7 48C79.8 304 0 383.8 0 482.3C0 498.7 13.3 512 29.7 512H418.3c16.4 0 29.7-13.3 29.7-29.7C448 383.8 368.2 304 269.7 304H178.3z"
            fill="currentColor"
          />
        </svg>
        <p
          class="namePath-0-2-21"
        >
          Feature 2
        </p>
      </a>
      <a
        class="menuLink-0-2-17"
        href="/feature3"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chart-pie icon-0-2-20"
          data-icon="chart-pie"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M304 240V16.6c0-9 7-16.6 16-16.6C443.7 0 544 100.3 544 224c0 9-7.6 16-16.6 16H304zM32 272C32 150.7 122.1 50.3 239 34.3c9.2-1.3 17 6.1 17 15.4V288L412.5 444.5c6.7 6.7 6.2 17.7-1.5 23.1C371.8 495.6 323.8 512 272 512C139.5 512 32 404.6 32 272zm526.4 16c9.3 0 16.6 7.8 15.4 17c-7.7 55.9-34.6 105.6-73.9 142.3c-6 5.6-15.4 5.2-21.2-.7L320 288H558.4z"
            fill="currentColor"
          />
        </svg>
        <p
          class="namePath-0-2-21"
        >
          Feature 3
        </p>
      </a>
      <a
        class="menuLink-0-2-17"
        href="/feature4"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-gear icon-0-2-20"
          data-icon="gear"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M495.9 166.6c3.2 8.7 .5 18.4-6.4 24.6l-43.3 39.4c1.1 8.3 1.7 16.8 1.7 25.4s-.6 17.1-1.7 25.4l43.3 39.4c6.9 6.2 9.6 15.9 6.4 24.6c-4.4 11.9-9.7 23.3-15.8 34.3l-4.7 8.1c-6.6 11-14 21.4-22.1 31.2c-5.9 7.2-15.7 9.6-24.5 6.8l-55.7-17.7c-13.4 10.3-28.2 18.9-44 25.4l-12.5 57.1c-2 9.1-9 16.3-18.2 17.8c-13.8 2.3-28 3.5-42.5 3.5s-28.7-1.2-42.5-3.5c-9.2-1.5-16.2-8.7-18.2-17.8l-12.5-57.1c-15.8-6.5-30.6-15.1-44-25.4L83.1 425.9c-8.8 2.8-18.6 .3-24.5-6.8c-8.1-9.8-15.5-20.2-22.1-31.2l-4.7-8.1c-6.1-11-11.4-22.4-15.8-34.3c-3.2-8.7-.5-18.4 6.4-24.6l43.3-39.4C64.6 273.1 64 264.6 64 256s.6-17.1 1.7-25.4L22.4 191.2c-6.9-6.2-9.6-15.9-6.4-24.6c4.4-11.9 9.7-23.3 15.8-34.3l4.7-8.1c6.6-11 14-21.4 22.1-31.2c5.9-7.2 15.7-9.6 24.5-6.8l55.7 17.7c13.4-10.3 28.2-18.9 44-25.4l12.5-57.1c2-9.1 9-16.3 18.2-17.8C227.3 1.2 241.5 0 256 0s28.7 1.2 42.5 3.5c9.2 1.5 16.2 8.7 18.2 17.8l12.5 57.1c15.8 6.5 30.6 15.1 44 25.4l55.7-17.7c8.8-2.8 18.6-.3 24.5 6.8c8.1 9.8 15.5 20.2 22.1 31.2l4.7 8.1c6.1 11 11.4 22.4 15.8 34.3zM256 336a80 80 0 1 0 0-160 80 80 0 1 0 0 160z"
            fill="currentColor"
          />
        </svg>
        <p
          class="namePath-0-2-21"
        >
          Feature 4
        </p>
      </a>
      <a
        class="menuLink-0-2-17"
        href="/administration"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-folder-minus icon-0-2-20"
          data-icon="folder-minus"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M448 480H64c-35.3 0-64-28.7-64-64V96C0 60.7 28.7 32 64 32H192c20.1 0 39.1 9.5 51.2 25.6l19.2 25.6c6 8.1 15.5 12.8 25.6 12.8H448c35.3 0 64 28.7 64 64V416c0 35.3-28.7 64-64 64zM184 272c-13.3 0-24 10.7-24 24s10.7 24 24 24H328c13.3 0 24-10.7 24-24s-10.7-24-24-24H184z"
            fill="currentColor"
          />
        </svg>
        <p
          class="namePath-0-2-21"
        >
          Administration
        </p>
      </a>
    </div>
    <div
      class="content-0-2-2"
    >
      <div
        class="title-0-2-3"
      >
        Connections
      </div>
      <div
        class="groupWrapper-0-2-4"
      >
        <p>
          Connections
        </p>
        <div>
          <button
            class="disableBtn-0-2-5"
          >
            Actions
            <svg
              fill="#ccc"
              height="10px"
              id="Layer_1"
              version="1.1"
              viewBox="0 0 386.257 386.257"
              width="10px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polygon
                points="0,96.879 193.129,289.379 386.257,96.879 "
              />
            </svg>
          </button>
          <button>
            Create Connection
          </button>
        </div>
      </div>
      <div
        class="contentWrapper-0-2-6"
      >
        <div
          class="wrapTable-0-2-7"
        >
          <table
            class="table-0-2-8"
          >
            <thead>
              <tr>
                <th />
                <th>
                  <button
                    class="wrapper-0-2-22"
                  >
                    <p>
                      Connection Name
                    </p>
                    <div>
                      <svg
                        fill="#ccc"
                        height="10px"
                        id="Layer_1"
                        version="1.1"
                        viewBox="0 0 386.257 386.257"
                        width="10px"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <polygon
                          points="0,96.879 193.129,289.379 386.257,96.879 "
                        />
                      </svg>
                    </div>
                  </button>
                </th>
                <th>
                  <button
                    class="wrapper-0-2-22"
                  >
                    <p>
                      Connection Type
                    </p>
                    <div>
                      <svg
                        fill="#ccc"
                        height="10px"
                        id="Layer_1"
                        version="1.1"
                        viewBox="0 0 386.257 386.257"
                        width="10px"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <polygon
                          points="0,96.879 193.129,289.379 386.257,96.879 "
                        />
                      </svg>
                    </div>
                  </button>
                </th>
                <th>
                  <button
                    class="wrapper-0-2-22"
                  >
                    <p>
                      Status
                    </p>
                    <div>
                      <svg
                        fill="#ccc"
                        height="10px"
                        id="Layer_1"
                        version="1.1"
                        viewBox="0 0 386.257 386.257"
                        width="10px"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <polygon
                          points="0,96.879 193.129,289.379 386.257,96.879 "
                        />
                      </svg>
                    </div>
                  </button>
                </th>
                <th>
                  <button
                    class="wrapper-0-2-22"
                  >
                    <p>
                      Created By
                    </p>
                    <div>
                      <svg
                        fill="#ccc"
                        height="10px"
                        id="Layer_1"
                        version="1.1"
                        viewBox="0 0 386.257 386.257"
                        width="10px"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <polygon
                          points="0,96.879 193.129,289.379 386.257,96.879 "
                        />
                      </svg>
                    </div>
                  </button>
                </th>
                <th>
                  <button
                    class="wrapper-0-2-22"
                  >
                    <p>
                      Creation Time
                    </p>
                    <div>
                      <svg
                        fill="#ccc"
                        height="10px"
                        id="Layer_1"
                        version="1.1"
                        viewBox="0 0 386.257 386.257"
                        width="10px"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <polygon
                          points="0,96.879 193.129,289.379 386.257,96.879 "
                        />
                      </svg>
                    </div>
                  </button>
                </th>
                <th>
                  <button
                    class="wrapper-0-2-22"
                  >
                    <p>
                      Last Updated
                    </p>
                    <div>
                      <svg
                        fill="#ccc"
                        height="10px"
                        id="Layer_1"
                        version="1.1"
                        viewBox="0 0 386.257 386.257"
                        width="10px"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <polygon
                          points="0,96.879 193.129,289.379 386.257,96.879 "
                        />
                      </svg>
                    </div>
                  </button>
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  <input
                    name="check"
                    type="radio"
                  />
                </td>
                <td
                  class="textPrimary"
                >
                   W4
                </td>
                <td>
                  Oracle
                </td>
                <td
                  class="success"
                >
                  Succeeded
                </td>
                <td>
                  vi.tran
                </td>
                <td>
                  2/5/2022-10:00 am
                </td>
                <td>
                  2/5/2022-10:00 am
                </td>
              </tr>
              <tr>
                <td>
                  <input
                    name="check"
                    type="radio"
                  />
                </td>
                <td
                  class="textPrimary"
                >
                  T24
                </td>
                <td>
                  Oracle
                </td>
                <td
                  class="fail"
                >
                  Failed
                </td>
                <td>
                  vi.tran
                </td>
                <td>
                  2/5/2022-10:00 am
                </td>
                <td>
                  2/5/2022-10:00 am
                </td>
              </tr>
              <tr>
                <td>
                  <input
                    name="check"
                    type="radio"
                  />
                </td>
                <td
                  class="textPrimary"
                >
                  T24
                </td>
                <td>
                  Oracle
                </td>
                <td
                  class=""
                >
                  N/A
                </td>
                <td>
                  vi.tran
                </td>
                <td>
                  2/5/2022-10:00 am
                </td>
                <td>
                  2/5/2022-10:00 am
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <nav
        aria-label="pagination navigation"
        class="MuiPagination-root MuiPagination-outlined pagination-0-2-9 css-1oj2twp-MuiPagination-root"
      >
        <ul
          class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
        >
          <li>
            <button
              aria-label="Go to previous page"
              class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary Mui-disabled MuiPaginationItem-previousNext css-r93niq-MuiButtonBase-root-MuiPaginationItem-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                data-testid="NavigateBeforeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
            </button>
          </li>
          <li>
            <button
              aria-current="true"
              aria-label="page 1"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary Mui-selected MuiPaginationItem-page css-r93niq-MuiButtonBase-root-MuiPaginationItem-root"
              tabindex="0"
              type="button"
            >
              1
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </li>
          <li>
            <button
              aria-label="Go to page 2"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary MuiPaginationItem-page css-r93niq-MuiButtonBase-root-MuiPaginationItem-root"
              tabindex="0"
              type="button"
            >
              2
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </li>
          <li>
            <button
              aria-label="Go to page 3"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary MuiPaginationItem-page css-r93niq-MuiButtonBase-root-MuiPaginationItem-root"
              tabindex="0"
              type="button"
            >
              3
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </li>
          <li>
            <button
              aria-label="Go to page 4"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary MuiPaginationItem-page css-r93niq-MuiButtonBase-root-MuiPaginationItem-root"
              tabindex="0"
              type="button"
            >
              4
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </li>
          <li>
            <button
              aria-label="Go to page 5"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary MuiPaginationItem-page css-r93niq-MuiButtonBase-root-MuiPaginationItem-root"
              tabindex="0"
              type="button"
            >
              5
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </li>
          <li>
            <div
              class="MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary MuiPaginationItem-ellipsis css-1v2lvtn-MuiPaginationItem-root"
            >
              …
            </div>
          </li>
          <li>
            <button
              aria-label="Go to page 10"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary MuiPaginationItem-page css-r93niq-MuiButtonBase-root-MuiPaginationItem-root"
              tabindex="0"
              type="button"
            >
              10
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </li>
          <li>
            <button
              aria-label="Go to next page"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary MuiPaginationItem-previousNext css-r93niq-MuiButtonBase-root-MuiPaginationItem-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                data-testid="NavigateNextIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </li>
        </ul>
      </nav>
    </div>
  </div>
  <footer
    class="wrapper-0-2-23"
  />
</DocumentFragment>
`;
